types/node/index.d.ts

```diff
@@ -1198,8 +1198,14 @@ declare module "http" {
         createConnection?: (options: ClientRequestArgs, oncreate: (err: Error, socket: net.Socket) => void) => net.Socket;
     }
 
+    interface ServerOptions {
+        IncomingMessage: any;
+        ServerResponse : any;
+    }
+
     class Server extends net.Server {
         constructor(requestListener?: (req: IncomingMessage, res: ServerResponse) => void);
+        constructor(options: ServerOptions, requestListener?: (req: IncomingMessage, res: ServerResponse) => void);
```

Comment:
I have to ask, are these part of the public API and/or considered stable?

Reply:
Yes they are but the `any` should be replaced by something more concrete. But it seems [my proposal](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/31390#issuecomment-447345172) is not working as is and needs some tuning. Will take a look once I find some time.

Reply:
Hey, the possibility to add this options to the server (custom request and response classes) was added in node 9, and node 10 is the current lts release, so it should be stable.

Reply:
Thanks @Flarna!